modules/renter/files.go

```diff
@@ -107,6 +108,34 @@ func (f *file) uploadProgress() float64 {
 	return 100 * (float64(uploaded) / float64(desired))
 }
 
+// redundancy returns the redundancy of the least redundant chunk. A file
+// becomes available when this redundancy is >= 1. Assumes that every piece is
+// unique within a file contract. -1 is returned if the file has size 0.
+func (f *file) redundancy() float64 {
+	if f.size == 0 {
+		return -1
+	}
+	piecesPerChunk := make([]int, f.numChunks())
+	// We cannot get the redundancy if piecesPerChunk has 0 length because later
+	// in the function we use the first element of the slice.
+	if len(piecesPerChunk) == 0 {
+		build.Critical("cannot get redundancy of a file with 0 chunks")
```

Comment:
The line a few above this one does return -1 for 0 sized files. I'll change this block to also return -1 instead of 0 to match the behavior of the block above.

Actually I was about to change it to return NaN to match the behavior of uploadProgress for 0 sized files, but that's another PR.